### PR TITLE
[WIP] Lots more analytics tracking.

### PR DESCRIPTION
**Pull request by darkruby501**
So far have added listContext to all instances of postItems I could think. Issues remain.

### CODE REVIEW

File: packages/lesswrong/components/posts/AllPostsPage.jsx [hunk 1]
```diff
@@ -102,16 +105,22 @@ class AllPostsPage extends Component {
...
-        postListParameters={postListParameters}
-        numTimeBlocks={numTimeBlocks}
-        dimWhenLoading={showSettings}
-        after={query.after || getAfterDefault({numTimeBlocks, timeBlock, timezone})}
-        before={query.before  || getBeforeDefault({timeBlock, timezone})}
-        reverse={query.reverse === "true"}
-        displayShortform={query.includeShortform !== "false"}
-      />
+      <AnalyticsContext
+          listContext={"allPostsPage"}
```
Comment (jimrandomh) [inline packages/lesswrong/components/posts/AllPostsPage.jsx]: Indentation nitpick: the preferred style when you have props that you want to split onto multiple lines looks like

```
<AnalyticsContext
  listContext="allPostsPage"
>
```

with the closing `>` on its own line, and the props indented one level.

**Comment by darkruby501** [comment]
This is ready for review to commence, only a few things remaining:

- Address issues Jim raised on the last PR.
- Change some of the context labels so as to have an overall consistent context labeling hierarchy. I'm thinking something like: pageContext -> pageSectionContext -> ElementContext, and separately ListContext, maybe I also want pageSubSectionContext? Not sure yet, but should be consistent.
- Remove console.log statement used in development.

File: packages/lesswrong/lib/reactRouterWrapper.jsx [hunk 2]
```diff
@@ -4,13 +4,17 @@ export const Link = reactRouter3.Link;
 
 export const withRouter = reactRouter3.withRouter;*/
 
-import React from 'react';
+import React, {useCallback} from 'react';
+import { useTracking } from '../lib/analyticsEvents';
+// packages/lesswrong/lib/reactRouterWrapper.jsx
```
Comment (darkruby501) [inline packages/lesswrong/lib/reactRouterWrapper.jsx]: need to remove these

**Comment by darkruby501** [comment]
Should also test properly on mobile, but so far, so good.

**Comment by darkruby501** [comment]
Things in this PR:
- rename reactRouterWrapper from .js to .jsx
- change the { Link } import for many files to use LW's version instead of default version, this causes Link tracking to work properly pretty much universally.
- Add lots of AnalyticsContext labels (i.e. wrapping code blocks) to the code for pages, sections of pages, elements, of page.
- Set up tracking of every Link element click.
- Add tracking to click handlers of tripleDot menus, notificationSection toggle, Navigation Bar toggle
- Add tracking to subscribeTo component.
- Cause all AnalyticsContexts to contain pathname of the page they're on.

**Comment by Discordius** [comment]
Seems overall fine, but needs to pass lint and address the DeepScan errors

**Pull request merged**